src/iuse.cpp

```diff
@@ -6459,20 +6559,800 @@ struct npc_photo_def : public JsonDeserializer, public JsonSerializer {
+            !g->m.sees( camera_pos, point_around_figure, dist + radius ) ||
+            ( ignored_points.find( point_around_figure ) != ignored_points.end() &&
+              !( point_around_figure == point && create_figure_desc ) ) ) {
+            continue; // disallow photos with not visible objects
+        }
+        units::volume volume_to_search = point_around_figure == bounds_center_point ? 0_ml :
+                                         min_visible_volume;
+
+        std::string furn_desc = colorized_feature_description_at( point_around_figure, item_found,
+                                volume_to_search );
+
+        const item item = get_top_item_at_point( point_around_figure, volume_to_search );
+        std::string item_name = colorized_item_name( item );
+
+        const optional_vpart_position veh_part_pos = g->m.veh_at( point_around_figure );
+        std::string unusual_ter_desc = colorized_ter_name_flags_at( point_around_figure,
+                                       camera_ter_whitelist_flags,
+                                       camera_ter_whitelist_types );
+        std::string ter_desc = colorized_ter_name_flags_at( point_around_figure );
+
+        const std::string trap_name = colorized_trap_name_at( point_around_figure );
+        const std::string field_desc = colorized_field_description_at( point_around_figure );
+
+        if( !furn_desc.empty() ) {
+            furn_desc = trap_name + furn_desc + field_desc;
+            if( point == point_around_figure && create_figure_desc ) {
+                description_furniture_on_figure = furn_desc;
+            } else {
+                ret_obj.furniture[ furn_desc ] ++;
+            }
+        } else if( veh_part_pos.has_value() ) {
+            const vehicle veh = veh_part_pos->vehicle();
+            const std::string veh_name = "<color_light_blue>" + veh.disp_name() + "</color>";
+            const std::string veh_hash = get_vehicle_hash( veh );
+
+            if( local_vehicles_recorded.find( veh_hash ) == local_vehicles_recorded.end() &&
+                point != point_around_figure ) {
+                ret_obj.vehicles[ veh_name ] ++;
+                vehicles_recorded.insert( veh_hash );
+            } else if( point == point_around_figure ) {
+                description_part_on_figure = veh_part_pos.part_displayed()->part().name() +
+                                             string_format( _( " from %s" ), veh_name );
+                if( ret_obj.vehicles.find( veh_name ) != ret_obj.vehicles.end() ) {
+                    ret_obj.vehicles.erase( veh_name );
+                }
+            }
+            local_vehicles_recorded.insert( veh_hash );
+        } else if( !item.is_null() ) {
+            item_name = trap_name + item_name + field_desc;
+            if( point == point_around_figure && create_figure_desc ) {
+                description_terrain_on_figure = ter_desc + _( " with a " ) + item_name;
```

Comment:
Translation these strings can be difficult. Image if a language requires the object (`item_name`) to come *before* the main word (`ter_desc`). Consider using `string_format` with placeholders instead. This allows the translators to change the word order as they need it.

Reply:
Done, but in some places it is still +. Should I replace more?